APISpec.md

```diff
@@ -0,0 +1,508 @@
+# Todo API Spec
+
+## API Resources
+
+* **User** - A User represents an owner of Todos and corresponds one-to-many to a Todo.
+
+* **Todo** - A Todo represents a task to be completed and corresponds many-to-one to a User.
+
+* **Session** - A Session represents an authenticated user session and corresponds many-to-one to a User.
+
+## Data Tables
+
+### Users:
+```
+ID - Guid (PK)
+Username - String
+Password - String (Hash)
+```
+
+### Todos:
+```
+ID - Guid (PK)
+UserID - Guid (FK - ID in Users)
+Title - String
+Status - Enumified String
+CreationDate - Datetime
+DueDate - Datetime
+```
+
+### Tokens:
+```
+ID - Guid (PK)
+UserID - Guid (FK - ID in Users)
+Token - String
+Expiration - Datetime
```

Comment:
You shouldn't need to track expiration in a table and the reason being is in most cases we implement sliding expiration for APIs where when a user logs in they receive a JWT in the form of a cookie (usually) and that cookie has an expiration value and every time an authenticated client they receive a new cookie with a new token whose expiration value is always X minutes in the future of the current time.

Reply:
You could also technically encode the expiration in a value in the JWT token itself (which is one of the advantages of JWT) as a ["user claim"](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social/additional-claims?view=aspnetcore-6.0) .